common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java

```diff
@@ -366,6 +369,72 @@ public void validateUpdateUserRequest(Request userRequest) {
     validateFrameworkDetails(userRequest);
   }
 
+  private void validateUserOrgField(Request userRequest) {
+    Map<String, Object> request = userRequest.getRequest();
+    if (StringUtils.isBlank((String) request.get(JsonKey.USER_ID))) {
+      ProjectCommonException.throwClientErrorException(
+          ResponseCode.mandatoryParamsMissing,
+          ProjectUtil.formatMessage(
+              ResponseCode.mandatoryParamsMissing.getErrorMessage(), JsonKey.USER_ID));
+    }
+    if (BooleanUtils.isTrue((Boolean) userRequest.getContext().get(JsonKey.PRIVATE))) {
+      if (request.containsKey(JsonKey.ORGANISATIONS)) {
+        if (!(request.get(JsonKey.ORGANISATIONS) instanceof List)) {
+          throwInvalidUserOrgData();
+        } else {
+          List<Object> list = (List<Object>) request.get(JsonKey.ORGANISATIONS);
+          for (Object map : list) {
+            if (!(map instanceof Map)) {
+              throwInvalidUserOrgData();
+            } else {
+              validateRole((Map<String, Object>) map);
+            }
+          }
+        }
+      }
+    } else {
+      if (request.containsKey(JsonKey.ORGANISATIONS)) {
+        ProjectCommonException.throwClientErrorException(
+            ResponseCode.errorUnsupportedField,
+            ProjectUtil.formatMessage(
+                ResponseCode.errorUnsupportedField.getErrorMessage(), JsonKey.ORGANISATIONS));
+      }
+    }
+  }
+
+  private void validateRole(Map<String, Object> map) {
```

Comment:
- validateRoles 

collect all roles sent by user into a Set<String> - then throw error for invalid roles - in this given set